### PR TITLE
Drop old syntax styles for givens

### PR DESCRIPTION
**Pull request by odersky**
* Drop `delegate`, `given ... as`, and `given` as an infix operator.
* Drop `DottyPredef.the`

### CODE REVIEW

File: library/src/dotty/DottyPredef.scala [hunk 1]
```diff
@@ -37,7 +37,5 @@ object DottyPredef {
     case ev: ValueOf[T] => ev.value
   }
 
-  inline def summon[T] given (x: T): x.type = x
-
-  inline def the[T] given (x: T): x.type = x
```
Comment (nicolasstucki) [inline library/src/dotty/DottyPredef.scala]: Add this removal to the description of the commit or PR

File: tests/neg/i4044a.scala [hunk 2]
```diff
@@ -8,7 +8,7 @@ def test(given QuoteContext) = {
     a // error
     $a
     '{$a} // error
-    '{ given QuoteContext = ???; '{$a} } // error
+    '{ given QuoteContext =???; '{$a} } // error
```
Comment (nicolasstucki) [inline tests/neg/i4044a.scala]: Why? Same for `tests/neg/i4044b.scala`

Comment (odersky) [reply tests/neg/i4044a.scala]: @nicolasstucki It's regex replacement breakages. There's simply too much to change - 2000 lines already! It's even too much to just inspect every line. 

Can you take this over, to fix the remaining problems? I am running out of time here.

Comment (nicolasstucki) [reply tests/neg/i4044a.scala]: Will fix them

File: docs/docs/reference/metaprogramming/macros-spec.md [hunk 3]
```diff
@@ -239,4 +239,4 @@ downright pleasant. A simple strategy first defines the underlying quoted or unq
 values using `Expr` and `Type` and then inserts quotes and splices to make the types
 line up. Phase consistency is at the same time a great guideline
 where to insert a splice or a quote and a vital sanity check that
-the result makes sense.
\ No newline at end of file
+the result makes sense.
```
Comment (anatoliykmetyuk) [inline docs/docs/reference/metaprogramming/macros-spec.md]: Why do the added and removed lines look indentical?

Comment (nicolasstucki) [reply docs/docs/reference/metaprogramming/macros-spec.md]: There was an extra space at the end of the line. If you select it you can see it.

Comment (anatoliykmetyuk) [reply docs/docs/reference/metaprogramming/macros-spec.md]: Strange, I can't see it when selecting the line. Also, Github highlights changes in spaces by default.

**Comment by milessabin** [comment]
shapeless updated in the community build here: https://github.com/lampepfl/dotty/pull/7244.

**Pull request merged**